homeassistant/components/netatmo/climate.py

```diff
@@ -125,44 +123,42 @@ async def async_setup_entry(
     data_handler = hass.data[DOMAIN][entry.entry_id][DATA_HANDLER]
 
     await data_handler.register_data_class(
-        HOMEDATA_DATA_CLASS_NAME, HOMEDATA_DATA_CLASS_NAME, None
+        CLIMATE_TOPOLOGY_CLASS_NAME, CLIMATE_TOPOLOGY_CLASS_NAME, None
     )
-    home_data = data_handler.data.get(HOMEDATA_DATA_CLASS_NAME)
+    climate_topology = data_handler.data.get(CLIMATE_TOPOLOGY_CLASS_NAME)
 
-    if not home_data or home_data.raw_data == {}:
+    if not climate_topology or climate_topology.raw_data == {}:
         raise PlatformNotReady
 
     entities = []
-    for home_id in get_all_home_ids(home_data):
-        for room_id in home_data.rooms[home_id]:
-            signal_name = f"{HOMESTATUS_DATA_CLASS_NAME}-{home_id}"
-            await data_handler.register_data_class(
-                HOMESTATUS_DATA_CLASS_NAME, signal_name, None, home_id=home_id
-            )
-            home_status = data_handler.data.get(signal_name)
-            if home_status and room_id in home_status.rooms:
-                entities.append(NetatmoThermostat(data_handler, home_id, room_id))
-
-    hass.data[DOMAIN][DATA_SCHEDULES].update(
-        update_climate_schedules(
-            home_ids=get_all_home_ids(home_data),
-            schedules=data_handler.data[HOMEDATA_DATA_CLASS_NAME].schedules,
+    for home_id in climate_topology.home_ids:
+        signal_name = f"{CLIMATE_STATE_CLASS_NAME}-{home_id}"
+        await data_handler.register_data_class(
+            CLIMATE_STATE_CLASS_NAME, signal_name, None, home_id=home_id
         )
-    )
+        climate_state = data_handler.data.get(signal_name)
```

Comment:
Use square brackets if you always know that it will be there.

```suggestion
        climate_state = data_handler.data.get[signal_name]
```